metrics/registry.go

```diff
@@ -0,0 +1,71 @@
+package metrics
+
+type registry struct {
+	provider Provider
+	SourceCh chan SourceOffset
+	SinkCh chan SinkOffset
+	PartitionCh chan PartitionInfo
+}
+
+type OffsetInfo struct {
+	Topic string
+	Partition int32
+	Offset int64
+}
+
+type PartitionInfo struct {
+	PartitionId int32
+	ConsumerId string
+	Topic string
+}
+
+type (
+	SinkOffset OffsetInfo
+	SourceOffset OffsetInfo
+)
+
+//Provider interface that implements metric registry types
+type Provider interface {
+	Init()
+	Ingest(interface{})
+}
+
+var (
+	Registry registry
+	MetricPort int
+)
+
+//Init creates a registry and initializes the metrics based on the registry type and implementation and returns the created registry
+func Init() registry{
+	pm := PrometheusMetrics{}
+
+	reg := registry{provider: &pm, SourceCh: make(chan SourceOffset, 1), SinkCh: make(chan SinkOffset, 1), PartitionCh: make(chan PartitionInfo)}
+	reg.provider.Init()
+	return reg
+}
+
+//TrackMetrics reads from various channels and calls ingest method updating the metrics
+func (reg *registry) TrackMetrics(){
```

Comment:
Where is this called from?

Reply:
This is method is called from main

Reply:
Dont see the caller

Reply:
Its being called from main only. It is being run as a separate go-routine "go metrics.Registry.TrackMetrics()"